src/Fabulous/Memo.fs

```diff
@@ -79,6 +80,13 @@ module Memo =
         {
             Key = MemoWidgetKey
             Name = "Memo"
+            GetTargetType =
+                fun widget ->
+                    let memoData = getMemoData widget
+                    let memoizedWidget = memoData.CreateWidget memoData.KeyData
```

Comment:
That is actually calling the function `memoData.CreateWidget memoData.KeyData`, which supposed to be slow (that's why it was memoized).

How often `GetTargetType` is called? If it is one of then it is probably Ok. Maybe we can just prohibit the use of Memo with VirtualCollections? But I'm not I understand all the implications, could you help me to understand a bit better?

Side note, I'm sorry if you already described it and I missed it, please point me to the doc if that is the case

Reply:
`GetTargetType` will be called every time a row needs to be displayed (fresh row or row reuse).

The way it works: XF will take the next data in the `IEnumerable` and ask `WidgetDataTemplateSelector` to give the corresponding `WidgetDataTemplate`, to be able to create (or reuse) a row.
To be able to create this `WidgetDataTemplate` (or reused cached one), `WidgetDataTemplateSelector` needs to call `GetTargetType` on the widget/memo.

So this can happen quite a lot.

What about storing the memoized widget target type inside `MemoData`?

Reply:
We don't know what it is prior actually calling the function :( So it is checking and egg problem